File: app/src/main/java/com/letianpai/robot/control/system/ServiceUtils.java
package com.letianpai.robot.control.system;

import android.app.ActivityManager;
import android.content.Context;
import android.text.TextUtils;
import android.util.Log;

import java.util.ArrayList;

/**
 * @author liujunbin
 */
public class ServiceUtils {
    /**
     * 判断服务是否开启
     *
     * @return
     */
    public static boolean isServiceRunning(Context context, String serviceName) {
        if (TextUtils.isEmpty(serviceName)) {
            return false;
        }

        ActivityManager myManager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
        ArrayList<ActivityManager.RunningServiceInfo> runningService = (ArrayList<ActivityManager.RunningServiceInfo>) myManager.getRunningServices(200);
        for (int i = 0; i < runningService.size(); i++) {
            if (runningService.get(i) != null && runningService.get(i).service != null && runningService.get(i).service.getClassName() != null) {
                if (runningService.get(i).service.getClassName().equals(serviceName)) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: app/src/main/java/com/letianpai/robot/control/system/ServiceUtils.kt
package com.letianpai.robot.control.system

import android.app.ActivityManager
import android.content.Context
import android.text.TextUtils

/**
 * @author liujunbin
 */
object ServiceUtils {
    /**
     * Determine if the service is on
     *
     * @return
     */
    fun isServiceRunning(context: Context, serviceName: String): Boolean {
        if (TextUtils.isEmpty(serviceName)) {
            return false
        }

        val myManager: ActivityManager =
            context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
        val runningService: ArrayList<ActivityManager.RunningServiceInfo?> =
            myManager.getRunningServices(200) as ArrayList<ActivityManager.RunningServiceInfo?>
        for (i in runningService.indices) {
            if (runningService[i] != null && runningService[i]!!.service != null) {
                if (runningService[i]!!.service.className == serviceName) {
                    return true
                }
            }
        }
        return false
    }
}
